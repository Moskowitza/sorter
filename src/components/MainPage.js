import React, { Component } from 'react';
import PropTypes from 'prop-types';
import Container from 'react-bootstrap/Container';
import Row from 'react-bootstrap/Row';
import Col from 'react-bootstrap/Col';
import CardList from './CardList';
import SearchBox from './SearchBox';
import Scroll from './Scroll';
import ErrorBoundary from './ErrorBoundary';
import './MainPage.css';


class MainPage extends Component {
  componentDidMount() {
    const { onRequestRobots } = this.props;
    onRequestRobots();
  }

  filteredRobots=() => {
    const { searchField, robots } = this.props;
    return robots.filter(robot => robot.name.toLowerCase().includes(searchField.toLowerCase()));
  }

  render() {
    const { isPending, error } = this.props;
    const { searchField, onSearchChange } = this.props;
    // eslint-disable-next-line max-len
    return isPending
      ? <h1>Loading</h1>
      : (
        <>
          <Container>
            <Row>
              <Col className="text-center">
                <h1 className="f1"> RoboFriends</h1>
              </Col>
            </Row>

            <SearchBox name="searchField" labelText="Search Robots" searchField={searchField} onChange={onSearchChange} />

          </Container>
          {error ? (<h2>{error}</h2>)
            : (
              <Scroll>
                <ErrorBoundary>
                  <Container fluid="true">
                    <CardList robots={this.filteredRobots()} />
                  </Container>
                </ErrorBoundary>
              </Scroll>
            )
          }
        </>
      );
  }
}
MainPage.defaultProps = {
  searchField: '',
  isPending: true,
  robots: [],
  error: '',
};
MainPage.propTypes = {
  searchField: PropTypes.string,
  error: PropTypes.string,
  isPending: PropTypes.bool,
  onSearchChange: PropTypes.func.isRequired,
  onRequestRobots: PropTypes.func.isRequired,
  robots: PropTypes.arrayOf(PropTypes.shape({
    name: PropTypes.string,
    email: PropTypes.string,
    id: PropTypes.number,
    address: PropTypes.shape({
      street: PropTypes.string,
      suite: PropTypes.string,
      city: PropTypes.string,
      zipcode: PropTypes.string,
      geo: PropTypes.shape({
        lat: PropTypes.string,
        lng: PropTypes.string,
      }),
    }),
    phone: PropTypes.string,
    website: PropTypes.string,
    company: PropTypes.shape({
      name: PropTypes.string,
      catchPhrase: PropTypes.string,
      bs: PropTypes.string,
    }),
  })),
};

export default MainPage;
