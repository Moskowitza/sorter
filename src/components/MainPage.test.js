import { shallow, mount, render } from 'enzyme';
import React from 'react';

// App is a connected Component, let's fix that
import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import MainPage from './MainPage';

configure({ adapter: new Adapter() });

describe('MainPage', () => {
  let wrapper;
  const mockProps = {
    onRequestRobots: jest.fn(),
    onSearchChange: jest.fn(),
    robots: [],
    searchField: '',
    isPending: false,
  };
  beforeEach(() => {
    wrapper = shallow(<MainPage {...mockProps} />).debug();
  });

  it('renders MainPage', () => {
    expect(wrapper).toMatchSnapshot();
  });
  it('filters correctly', () => {
    const mockProps2 = {
      onRequestRobots: jest.fn(),
      onSearchChange: jest.fn(),
      robots: [
        {
          id: 1,
          name: 'toki',
          email: 'toki@email.com',
        },
      ],
      searchField: 'a',
      isPending: false,
    };
    const wrapper2 = shallow(<MainPage {...mockProps2} />);
    expect(wrapper2.instance().filteredRobots()).toEqual([]);
  });
});
