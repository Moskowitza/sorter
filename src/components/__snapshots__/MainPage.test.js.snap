// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MainPage renders MainPage 1`] = `
"<Fragment>
  <Container fluid={false}>
    <Bootstrap(Row)>
      <Col className=\\"text-center\\">
        <h1 className=\\"f1\\">
           RoboFriends
        </h1>
      </Col>
    </Bootstrap(Row)>
    <SearchBox name=\\"searchField\\" labelText=\\"Search Robots\\" searchField=\\"\\" onChange={[Function: mockConstructor]} />
  </Container>
  <Scroll>
    <ErrorBoundary>
      <Container fluid=\\"true\\">
        <CardList robots={{...}} />
      </Container>
    </ErrorBoundary>
  </Scroll>
</Fragment>"
`;
